scraper/maiia/maiia_center_scrap.py

```diff
@@ -0,0 +1,106 @@
+        creneaux = list()
+        for schedule in schedules:
+            creneaux.append(f'{schedule["startTime"]}-{schedule["endTime"]}')
+        if creneaux:
+            business_hours[key] = ' '.join(creneaux)
+    return business_hours
+
+
+def maiia_center_to_csv(center: dict, root_center: dict) -> dict:
+    if 'url' not in center:
+        logger.warning(f'url not found - {center}')
+    csv = dict()
+    csv['gid'] = center['id'][:8]
+    csv['nom'] = center['name']
+    csv['rdv_site_web'] = f'{MAIIA_URL}{center["url"]}?centerid={center["id"]}'
+    if 'publicInformation' in center:
+        if 'address' in center['publicInformation']:
+            insee = center['publicInformation']['address'].get('inseeCode', '')
+            csv['com_insee'] = center['publicInformation']['address']['inseeCode']
+            if len(insee) < 5:
+                zip = center['publicInformation']['address']['zipCode']
+                csv['com_insee'] = departementUtils.cp_to_insee(zip)
+            csv['address'] = center['publicInformation']['address']['fullAddress']
+            if 'location' in center['publicInformation']['address']:
+                csv['long_coor1'] = center['publicInformation']['address']['location']['coordinates'][0]
+                csv['lat_coor1'] = center['publicInformation']['address']['location']['coordinates'][1]
+        if 'officeInformation' in center['publicInformation']:
+            csv['phone_number'] = center['publicInformation']['officeInformation'].get(
+                'phoneNumber', '')
+            if 'openingSchedules' in center['publicInformation']['officeInformation']:
+                csv['business_hours'] = maiia_schedule_to_business_hours(
+                    center['publicInformation']['officeInformation']['openingSchedules'])
+    return csv
+
+
+def main():
+    centers = list()
+    centers_ids = list()
+    logger.info('Starting Maiia centers download')
+
+    for speciality in MAIIA_SPECIALITIES:
+        logger.info(f'Fetching speciality {speciality}')
+        result = get_centers(speciality)
+        all_centers = list()
+        for center in result:
+            if center.get('type') != "CENTER":
+                continue
+            has_vaccinne_reason = False
+            for consultation_reason in center['consultationReasons']:
+                if 'injectionType' in consultation_reason and consultation_reason['injectionType'] in ['FIRST', 'SECOND']:
+                    has_vaccinne_reason = True
+            if has_vaccinne_reason:
+                if center['center']['id'] in centers_ids:
+                    continue
+                centers.append(maiia_center_to_csv(center['center'], center))
+                centers_ids.append(center['center']['id'])
+            for child_center in center['center']['childCenters']:
+                if child_center['speciality']['code'] == 'VAC01' and 'url' in child_center:
+                    if child_center['id'] in centers_ids:
```

Comment:
une raison particuliere pour ne pas ajouter un autre and et faire une if en plus?

Reply:
non je corrige